### ISSUE TITLE
Stop element dragging on certain event

### ISSUE THREAD

**Issue by monis01** [valor-software/ng2-dragula#902]
Hi 
I am using ng2-dragula for dragging different components, and it's look like 
```
<div  dragula="COMPONENTS">
    <div #addComponent></div>
</div>
```
`addComponent` is creating some dynamic components and in these dynamic components I have created a class named 'section-drag', i.e.
```
<a href="javascript:void(0);" class="btn grey section-drag">
   <i class="fa fa-arrows section-drag"></i>
</a>
``` 
and in my code 

```
dragulaService.createGroup("COMPONENTS", {
                moves: this.moveComponents
});
private moveComponents = (el, container, handle)=>{
        return handle.className.match(/section-drag/) != null;
}
```
So I require to stop dragging on a certain button click, so I did
```
<div *ngIf="someCondition">
   <a href="javascript:void(0);" class="btn grey section-drag">
      <i class="fa fa-arrows section-drag"></i>
   </a>
</div>
```
so I just remove it but still `this.moveComponents` function runs (and it should), but sometimes even on mentioning a class name (`section-drag`) element get's dragged. 

Is there any better way  If I could stop dragging or anything ON or Off in dragula service rather manipulating my drake element on a certain event.

Thanks

PS: I did not got anything similar in docs or might be I have missed something.

**Comment by cormacrelf** [comment]
#784 which was fixed in v2. Set `[dragula]="shouldDrag ? "VAMPIRES" : null"` and `(click)="shouldDrag = false"`

**Comment by monis01** [comment]
Hi
Thanks for the code, but sometimes it is seen I am able to drag the element completely from anywhere even my drag handle is defined. My dragula initialization is  

```
dragulaService.createGroup("COMPONENTS", {
      moves: this.moveComponents
});
private moveComponents = (el, container, handle)=>{
        return handle.className.match(/section-drag/) != null;
}
this.dragulaSubs.add(this.dragulaService.drop("COMPONENTS").subscribe(this.dropComponents))
private dropComponents = ({ name, el, target, source, sibling })=>{
//@ My whole code
}
```
sometimes it is dragged properly from class handler i.e. `section-drag`
![screenshot from 2018-09-20 20-00-29](https://user-images.githubusercontent.com/7036532/45825662-6071d780-bd10-11e8-8e0a-b7b7682d5e26.png)

and also I am able to debug my `moveComponents` function  but sometimes able to drag from anywhere and also cannot debug my `moveComponents` function.
![screenshot from 2018-09-20 20-03-23](https://user-images.githubusercontent.com/7036532/45825904-eb52d200-bd10-11e8-9bc4-2036a7efb73d.png)

I don't know how it is inconsistent.
Thanks

**Comment by monis01** [comment]
I have fixed that, I was missing few things here 
1. I was not unsubscribing the subscription that I used for drop function i.e.  `this.dragulaSubs.add(this.dragulaService.drop("COMPONENTS").subscribe(this.dropComponents))`
2. My application requires an auto-update so I need to destroy my component and run ` this.ngOnInit` after certain interval (and my service initialization code was in constructor function) so after destroying `this.dragulaService.destroy('COMPONENTS');` and unsubscribing subscription in `ngOnDestroy` when component is re-rendered all it gets  html with declaration like  `<div  dragula="COMPONENTS"></div>` and nothing in component, so I have moved my initialization  from constructor function to `ngOnInit()`. After destroying  I reinitialize it again and it solved the problem.
I believe irrespective of problem solved, this would be the way too for such cases.
Thanks
